src/java/org/jivesoftware/openfire/XMPPServer.java

```diff
@@ -175,6 +186,35 @@ public XMPPServerInfo getServerInfo() {
     }
 
     /**
+     * Returns an ACDF object, which can perform decisions.
+     *
+     * @return AccessControlDecisionFunction implementation
+     */
+    public AccessControlDecisionFunction getAccessControlDecisionFunction() {
+        return accessControlDecisionFunction;
+    }
+
+    /**
+     * Sets the AccessControlDecisionFunction implementation.
+     *
+     * @param acdf AccessControlDecisionFunction object.
+     */
+    public void setAccessControlDecisionFunction(AccessControlDecisionFunction acdf) {
+        this.accessControlDecisionFunction = acdf;
+        if (acdf == null) {
+            getIQDiscoInfoHandler().removeServerFeature("urn:xmpp:sec-label:0");
+            for (PacketInterceptor i : InterceptorManager.getInstance().getInterceptors()) {
+                if (i instanceof MessageInterceptor) {
```

Comment:
This would also unload third party / plugin implementations, I think. Is that intended?